src/lib/tools/fetchUser.ts

```diff
@@ -0,0 +1,46 @@
+import { tool } from "ai"
+import { metaphysics } from "../metaphysics"
+import { z } from "zod"
+
+export const fetchUser = tool({
+  description: `
+    Fetch the current user's record, which contains basic information about the user,
+    including their name, profession and location.
+  `,
+
+  parameters: z.object({}),
+
+  execute: async () => {
+    const accessToken = process.env.USER_ACCESS_TOKEN
+
+    const query = `query {
+      me {
+        name
+        initials
+        email
+        bio
+        profession
+        location {
+          internalID
+          city
+          summary
+          state
+          country
+          coordinates {
+            lat
+            lng
+          }
+        }
+      }
+    }`
+
+    const variables = {}
+
+    const headers = {
+      "X-ACCESS-TOKEN": accessToken!,
```

Comment:
I think this is fine for a demo repo/PR, but as we move towards more advances patterns that could end up in production (possibly), just noting we make sure to should be careful this non-null guarantee doesn't get copied over. Would be an easy one to miss.

Reply:
Yeah good call. Probably other improvements to make re: token handling as well, like it should maybe be passed in from the caller, rather than assumed to be in the env.